### PR TITLE
[IMP] mail: add message_type for outgoing emails

### PR DESCRIPTION
**Pull request by tde-banana-odoo**

### CODE REVIEW

**Review by awa-odoo — COMMENTED** [review]
🐟 !

Left some comments.

Seems like a surprisingly low amount of diff for that change, I hope your grep skills are strong 😄 

Cheers!

File: addons/mail/models/mail_message.py [hunk 1]
```diff
@@ -115,14 +115,20 @@ def default_get(self, fields):
     record_company_id = fields.Many2one('res.company', 'Company', ondelete='set null')
     # characteristics
     message_type = fields.Selection([
```
[the remark this answers is not in the document]
Comment (tde-banana-odoo) [reply addons/mail/models/mail_message.py:117]: Actually the new help us sufficient now that it is complete.

File: addons/mail/models/discuss/discuss_channel.py [hunk 2]
```diff
@@ -468,8 +468,8 @@ def _notify_get_recipients(self, message, msg_vals, **kwargs):
         message_type = msg_vals.get('message_type', 'email') if msg_vals else message.message_type
         pids = msg_vals.get('partner_ids', []) if msg_vals else message.partner_ids.ids
 
-        # notify only user input (comment or incoming emails)
-        if message_type not in ('comment', 'email'):
+        # notify only user input (comment or incoming / outgoing emails)
+        if message_type not in ('comment', 'email', 'email_outgoing'):
```
[the remark this answers is not in the document]
Comment (tde-banana-odoo) [reply addons/mail/models/discuss/discuss_channel.py:472]: Don't really know ... at least it is backward compatible.

File: addons/mail/views/mail_mail_views.xml [hunk 3]
```diff
@@ -124,9 +124,14 @@
...
                     <filter name="exception" string="Failed" domain="[('state','=','exception')]"/>
                     <separator/>
-                    <filter name="type_email" string="Email" domain="[('message_type','=','email')]"/>
-                    <filter name="type_comment" string="Comment" domain="[('message_type','=','comment')]"/>
-                    <filter name="type_notification" string="Notification" domain="[('message_type','=','notification')]"/>
+                    <filter name="filter_type_email_outgoing" string="Outgoing Email"
+                            domain="[('message_type','=','email_outgoing')]"/>
+                    <filter name="filter_type_comment" string="Comment"
+                            domain="[('message_type','=','comment')]"/>
+                    <filter name="filter_type_email" string="Incoming Email"
```
[the remark this answers is not in the document]
Comment (tde-banana-odoo) [reply addons/mail/views/mail_mail_views.xml:129]: Not really. Main usage of this view is to find outgoing emails: they are generally based on 'outgoing email' as they are generated by mailings. Then you have 'comments' or 'incoming emails' that post a message and then could generate emails, but I think the main usage is: where are my mailing emails. So I put outgoing as first, then the others, can put incoming second.

File: addons/sms/models/mail_message.py [hunk 4]
```diff
@@ -14,9 +14,9 @@ class MailMessage(models.Model):
     gateway. """
     _inherit = 'mail.message'
 
-    message_type = fields.Selection(selection_add=[
-        ('sms', 'SMS')
-    ], ondelete={'sms': lambda recs: recs.write({'message_type': 'email'})})
+    message_type = fields.Selection(
+        selection_add=[('sms', 'SMS')],
+        ondelete={'sms': lambda recs: recs.write({'message_type': 'comment'})})
```
[the remark this answers is not in the document]
Comment (tde-banana-odoo) [reply addons/sms/models/mail_message.py:19]: The previous one was not correct, we should consider posted sms are comments actually, should be the default value when we have no other clue. Can add a note in commit message.

**Review by tde-banana-odoo — COMMENTED** [review]
Zblip, answers, and added some new changes, notably a better default value for mail.mail / mail.message message_type

**Comment by tde-banana-odoo** [comment]
@robodoo r+

**Comment by MiquelRForgeFlow** [comment]
Sorry if I bother, but, may I ask how can we know in older versions if a mail was created from mailgateway (incoming) or if it was an outgoing mail?

**Pull request closed without merging**